code/modules/reagents/chemistry/reagents/rechem.dm

```diff
@@ -0,0 +1,147 @@
+
+GLOBAL_LIST_EMPTY(bluestreams)
+
+/datum/reagent/quaprotium
+	name = "Quanprotium"
+	description = "A complex and rare chemical mix that has unique bluespace properties"
+	taste_description = "your tongue licking something through other dimensions" //Some memes *_*
+	color = "#2b78ed" //RGB: 43, 120, 237
+	can_synth = FALSE //Would be too OP
+	pH = 11
+
+	metabolization_rate = REAGENTS_METABOLISM * 1.5
+
+/datum/reagent/quaprotium/expose_turf(turf/exposed_turf, reac_volume)
+	. = ..()
+	if(reac_volume > 5 && !locate(/obj/effect/bluespace_stream) in exposed_turf)
+		new /obj/effect/bluespace_stream(exposed_turf)
+
+/datum/reagent/quaprotium/on_mob_life(mob/living/carbon/M) //Affects living beings in a same way as bluespace crystals
+	if(current_cycle > 10 && prob(15))
+		to_chat(M, "<span class='warning'>You feel unstable...</span>")
+		M.Jitter(2)
+		current_cycle = 1
+		addtimer(CALLBACK(M, /mob/living/proc/bluespace_shuffle), 30)
+	. = ..()
+
+/obj/effect/bluespace_stream
+	name = "bluespace stream"
+	desc = "A strange, previously hidden portal through bluespace. It might be a bad idea."
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "bluestream"
+	layer = ABOVE_MOB_LAYER
+
+/obj/effect/bluespace_stream/Initialize()
+	. = ..()
+	GLOB.bluestreams.Add(src)
+	new /obj/effect/temp_visual/bluespace_fissure(get_turf(src))
+
+/obj/effect/bluespace_stream/attack_hand(mob/user)
+	var/obj/effect/bluespace_stream/linked = pick(GLOB.bluestreams - src)
+	if(!linked)
+		return
+	var/slip_in_message = pick("slides sideways in an odd way, and disappears", "jumps into an unseen dimension",\
+		"sticks one leg straight out, wiggles [user.p_their()] foot, and is suddenly gone", "stops, then blinks out of reality", \
+		"is pulled into an invisible vortex, vanishing from sight")
+	var/slip_out_message = pick("silently fades in", "leaps out of thin air","appears", "walks out of an invisible doorway",\
+		"slides out of a fold in spacetime")
```

Comment:
Please consider changing the flavor text of the portals, these were written with the idea that onlookers would see someone acting weirdly and then suddenly disappearing in an invisible fold of bluespace, but don't make much sense if the portal is just visible and accessible to everyone

Reply:
Alternatively, i'd suggest using portal code from the hand-tele or the portal gun, since they're coded more extensively and support stuff like pushing stuff through them or teleporting projectiles

Reply:
Alrighty, will change the flavors. However, this is intended to be human-only AND manual portals because I don't want any kinds of meme traps with it